### PR TITLE
mesos: install python bindings

### PR DESCRIPTION
**Pull request by tdsmith**
* installs python bindings and dependencies
* installs ec2 scripts into share/mesos/ec2

The mesos build scripts should build these but they work in an idiosyncratic way and don't seem to produce correct results.

Previously: https://github.com/Homebrew/homebrew/pull/37087#issuecomment-77114099

### CODE REVIEW

File: Library/Formula/mesos.rb [hunk 1]
```diff
@@ -65,5 +139,11 @@ def install
     Process.kill("TERM", master)
     Process.kill("TERM", slave)
     assert File.exist?("#{testpath}/executed")
+
+    user_site = Language::Python.user_site_packages("python")
+    mkdir_p user_site
+    pth_contents = "import site; site.addsitedir('#{Language::Python.homebrew_site_packages}')\n"
+    (user_site/"homebrew.pth").atomic_write pth_contents
+    system "python", "-c", "import mesos.native"
```
Comment (xu-cheng) [inline Library/Formula/mesos.rb]: I think test case should not change any file outside test path.

Comment (tdsmith) [inline Library/Formula/mesos.rb]: I agree. It looks like this touches a user's files, but it doesn't; $HOME is set to a temporary path for tests and Python respects it.

File: Library/Formula/mesos.rb [hunk 2]
```diff
@@ -40,9 +88,35 @@ def install
 
     ENV.cxx11
 
-    system "./configure", *args
+    system "./configure", *(args + ["--disable-python"])
```
Comment (mikemcquaid) [inline Library/Formula/mesos.rb]: Can we add this to the array instead?

File: Library/Formula/mesos.rb [hunk 3]
```diff
@@ -15,22 +15,70 @@ class Mesos < Formula
...
   needs :cxx11
 
   def install
+    boto_path = libexec/"boto/lib/python2.7/site-packages"
+    ENV.prepend_create_path "PYTHONPATH", boto_path
+    resource("boto").stage do
+      system "python", *Language::Python.setup_install_args(libexec/"boto")
+    end
+    (lib/"python2.7/site-packages").mkpath
+    (lib/"python2.7/site-packages/homebrew-mesos-boto.pth").atomic_write boto_path + "\n"
```
Comment (mikemcquaid) [inline Library/Formula/mesos.rb]: `"#{boto_path}\n"` is maybe nicer?

Comment (mikemcquaid) [inline Library/Formula/mesos.rb]: Also, can we just use `.write` here? I assume not but I'm maybe missing something.

File: Library/Formula/mesos.rb (hunk 2 above)
Comment (mikemcquaid) [inline Library/Formula/mesos.rb]: Or, looking at below, have `python_args` and `no_python_args` or something?

Comment (tdsmith) [inline Library/Formula/mesos.rb]: Occurs I could also just pass the extra --en/disable-python argument in
front of *args; is that okay?

Comment (mikemcquaid) [inline Library/Formula/mesos.rb]: Yeh, that would be nicer :+1:

**Pull request closed without merging**